sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/XmlExternalEntityShouldNotBeParsed_XmlReader_Net452.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Xml;
+
+namespace SonarAnalyzer.UnitTest.TestCases
```

Comment:
in one of these files, it would be good to add some test cases similar to what I did in the `VariousUnsafeCombinations` (well, what would make sense for this rule)

**1**. document what's happening when the XmlReaderSettings is just a parameter of a factory method
---> what happens if nothing is set on the settings
---> what happens when something is set (so settings are initialized w/o constructor)

**2**. from [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmlreader?view=netframework-4.8#creating-an-xml-reader), what happens when you use the XmlReader `Create` constructor called from one of the inheriting classes, such as
```
            XmlNodeReader.Create("", settings, "resources/");  // returns XmlReader
```
or 
```
            XmlTextReader.Create("", settings, "resources/"); // returns XmlReader
```

Reply:
Good point I've also found a bug with this because fields were not correctly handled :). 
I've added now a few additional tests similar with what has been done for XmlDocument and XmlTextReader.